presto-hive/src/main/java/io/prestosql/plugin/hive/HiveSessionProperties.java

```diff
@@ -123,6 +123,9 @@ public HiveSessionProperties(
             ParquetReaderConfig parquetReaderConfig,
             ParquetWriterConfig parquetWriterConfig)
     {
+        InsertExistingPartitionsBehavior insertExistingPartitionsBehavior = hiveConfig.getInsertExistingPartitionsBehavior();
+        InsertExistingPartitionsBehavior.validate(insertExistingPartitionsBehavior, hiveConfig.isImmutablePartitions());
```

Comment:
Why is `@PostConstruct` validation not enough

Reply:
Good point! Removed.